src/binaryen-c.cpp

```diff
@@ -3914,35 +4013,84 @@ uint32_t BinaryenGetMemorySegmentByteOffset(BinaryenModuleRef module,
   return 0;
 }
 bool BinaryenHasMemory(BinaryenModuleRef module) {
-  return ((Module*)module)->memory.exists;
+  return !((Module*)module)->memories.empty();
 }
-BinaryenIndex BinaryenMemoryGetInitial(BinaryenModuleRef module) {
-  return ((Module*)module)->memory.initial;
+BinaryenIndex BinaryenMemoryGetInitial(BinaryenModuleRef module,
+                                       const char* name) {
+  // Maintaining compatibility for instructions with a single memory
+  if (name == nullptr && module->memories.size() == 1) {
+    name = module->memories[0]->name.c_str();
+  }
+  auto* memory = ((Module*)module)->getMemoryOrNull(name);
+  if (memory == nullptr) {
+    Fatal() << "invalid memory '" << name << "'.";
+  }
```

Comment:
Since we are producing a fatal error anyway, might as well call `getMemory` instead of `getMemoryOrNull` and skip the error checking here.